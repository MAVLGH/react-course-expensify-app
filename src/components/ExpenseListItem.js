import React from 'react';
import { NavLink } from 'react-router-dom';
import moment from 'moment';

const ExpenseListItem = ({ description, amount, createdAt, id }) => (
    <div>
        <h3><NavLink to={`/edit/${id}`} activeClassName="is-active" exact={true}>{description}</NavLink></h3>
        <p>Amount: {`USD$${(amount / 100)} `}-{` Date: ${moment(createdAt).format('DD/MM/YYYY HH:mm:ss').toString()}`}</p>
    </div>
);

export default ExpenseListItem;

