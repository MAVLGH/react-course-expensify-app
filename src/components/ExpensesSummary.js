import React from 'react';
import { connect } from 'react-redux';
import numeral from 'numeral';
import totalExpensesSelector from '../selectors/expenses-total'

export class ExpensesSummary extends React.Component {
  constructor(props) {
    super(props);
    this.expensesLength = this.props.expenses.length
    this.expensesTotal = `USD${numeral(totalExpensesSelector(this.props.expenses) / 100).format('$0,0.00')} `
  }
  render() {
    return (
      <div>
        <h1>Expenses Summary</h1>
        {this.props.expenses.length>0 ? <p>Viewing {this.expensesLength} totalling {this.expensesTotal}</p> : <p>No summary</p>}
      </div>
    )
  }
}

const mapStateToProps = (state, props) => ({
  expenses: state.expenses
}
)

export default connect(mapStateToProps)(ExpensesSummary)