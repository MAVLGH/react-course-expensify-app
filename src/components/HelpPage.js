import React from 'react'

const HelpPage = () => (
    <div>
        This is from my expense Help component
    </div>
);

export default HelpPage;