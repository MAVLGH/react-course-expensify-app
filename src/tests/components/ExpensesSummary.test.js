import React from 'react'
import expenses from '../fixtures/expenses'
import { shallow } from 'enzyme'
import { ExpensesSummary } from '../../components/ExpensesSummary'

test('should render Summary with no expenses correctly', () => {
  const wrapper = shallow(<ExpensesSummary expenses={[]}/>)
  expect(wrapper).toMatchSnapshot();
})

test('should render Summary with expenses correctly', () => {
  const wrapper = shallow(<ExpensesSummary expenses={expenses}/>)
  expect(wrapper).toMatchSnapshot();
})