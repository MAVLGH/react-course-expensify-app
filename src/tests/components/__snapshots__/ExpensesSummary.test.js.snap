// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render Summary with expenses correctly 1`] = `
<div>
  <h1>
    Expenses Summary
  </h1>
  <p>
    Viewing 
    3
     totalling 
    USD$1,141.95 
  </p>
</div>
`;

exports[`should render Summary with no expenses correctly 1`] = `
<div>
  <h1>
    Expenses Summary
  </h1>
  <p>
    No summary
  </p>
</div>
`;
